src/nixpacks/plan/mod.rs

```diff
@@ -46,6 +49,8 @@ pub struct BuildPlan {
 
     pub phases: Option<Phases>,
 
+    pub pinned_archive: Option<String>,
```

Comment:
This is not necessary as phases already support setting the Nix archive to a specific version, https://github.com/railwayapp/nixpacks/blob/7653b2f1a8b7c4532adb51d1c8f7f20cbd26386c/src/nixpacks/plan/phase.rs/#L207.

See [this](https://github.com/railwayapp/nixpacks/blob/7653b2f1a8b7c4532adb51d1c8f7f20cbd26386c/src/providers/go.rs/#L65) as an example